confmap/confmap.go

```diff
@@ -133,6 +133,10 @@ func sanitizeExpanded(a any, useOriginal bool) any {
 		return c
 	case []any:
 		var newSlice []any
+		if reflect.ValueOf(m).IsNil() {
```

Comment:
why do you need to check via reflect package, and not just `m == nil`?

Reply:
@yurishkuro 
- We create a nil slice here: https://github.com/go-viper/mapstructure/blob/main/mapstructure.go#L496
- That nil slice gets turned into a concrete value in https://github.com/go-viper/mapstructure/blob/main/decode_hooks.go#L95 which gets passed in https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/confmap.go#L290
As a result of this, we need to dig into the type using reflection to see if it is nil. Let me know if you see an alternative here.

Reply:
done